### DISCUSSION TITLE
glXF Support

### DISCUSSION THREAD

**Discussion by m11m** [donmccurdy/glTF-Transform#1090] [Ideas]
https://github.com/KhronosGroup/glXF

> The glTF Experience Format (glXF) specification defines a new JSON format that contains references to glTF 2.0 assets, a node hierarchy, and optional extensions for interactivity, augmented reality anchoring, or other capabilities.

> A glXF file is essentially a glTF file that does not specify any geometry, animations, textures, etc. directly, but rather incorporates them by reference to other glTF files. glXF files may incorporate other glXF files by reference, forming an acyclic graph of incorporation, but all geometry comes, in the end, from glTFs as the leaf nodes of that graph.

I don't know of anywhere that uses glXF currently, but this repo was created by Khronos and I think its an interesting idea. I know its something I'd like to experiment with. Would it make sense for glTF-Transform to support glXF files at some point, as internally its mostly just a glTF file?

**Comment by donmccurdy** [comment]
I'm not sure whether glXF is something Khronos is actively developing, but assuming so, I think the two main parts are:

1. Support for references from .glxf → .gltf
2. Support for whichever extensions are ratified for .glxf

_Theoretically_, both of these would make sense for glTF Transform to support, and glTF Transform feels like a good technical fit for working with this data.

_Practically_, glXF might be outside my complexity budget for the project at this time. Extensions like general property animation and behavior graphs have larger effects on how all the optimizations in the `@gltf-transform/functions` module should work. References similarly require some decisions about how to deal with these resources, and whether it's still reasonable to load the entire asset in-memory at that scale.

I'll try to write something more general about the complexity budget constraints, when I can. But basically it comes down to needing more time to work on the project, more contributors, or funding to buy one or both of those things.

**Reply by m11m** [reply]
Totally understand the budget / time constraints. Not sure either if Khronos is still interested in it, they seem to work behind closed doors so I can never tell. I don't need glXF at the moment but it may be pretty useful in the future for a project I'm working on, so would love to help out with glXF support when the time comes (through contribution, would also love to give money if I have enough at the time).

And thanks for all the work you do in the ecosystem.

**Reply by donmccurdy** [reply]
Thank you @kayhhh! I went ahead and filed an issue to track this, since I expect it will come up again: https://github.com/donmccurdy/glTF-Transform/issues/1138

For now the status is the same as described above, though.
